src/object/srv_obj.c

```diff
@@ -1296,8 +1298,16 @@ ds_obj_rw_handler(crt_rpc_t *rpc)
 	/* Stop the distribute transaction */
 	rc = dtx_leader_end(&dlh, cont, rc);
 	if (rc == -DER_AGAIN) {
-		flags |= ORF_RESEND;
-		D_GOTO(again, rc);
+		if (!dlh.dlh_handle.dth_renew) {
+			flags |= ORF_RESEND;
+			D_GOTO(again, rc);
```

Comment:
does the old one get aborted in such case?  I'm thinking of case where one replica reports -DER_AGAIN but another didn't have first dtx yet so it did not and therefore has this one in the tree.   Does it get aborted in such case?

Reply:
I hit -DER_AGAIN during daos_racer test. I think it was caused by DAOS-3882 which root reason was not found yet at that time. But current DTX logic does not handle DER_AGAIN properly as to it will access some released dth_handle, so I made the patch.

Now, with the patch DAOS-3882 landed to master, DER_AGAIN may be hidden temporarily, but DTX related logic issues should be fixed.

Reply:
As for the old DTX for DER_AGAIN/DER_INPROGRESS case, the dtx_leader_end() has already aborted them.